README.md

```diff
@@ -140,6 +159,45 @@ import { utilName } from 'contexture-client'
 | `hasValue` | An internal utility that checks if a node has a value and isn't in an error state. Used primarily in custom reactors to help figure out if other nodes would be affected by an event. |
 | `exampleTypes` | A set of example types. This will likely be split out in a future version to its own repo. |
 | `mockService` | Useful for mocking services during testing. Takes a config object of logInput, logOutput, and a mocks function which will should return results for a node as input (defaults to fixed results by type) and returns a function for use as a `service` for a contexture-client instance. |
+| `subquery` | A tool for creating a subquery. See the section below. |
+
+
+#### Subquery
+A subquery (in contexture-client) is about taking the output of one search and makng it the input for another search.
+This is an in memory, cross-database, "select in" join on sources that don't need to be relational.
+
+This works by providing a source node from which to **get** values, and a target to **use** those values.
+
+The client exposes a method to create subqueries between two trees as a top level export, with this signature:
+`(types, from, fromPath, to, toPath)`
+It takes `types` (just like the client itself), the tree and path of the source node, and then the tree and path of the target node.
+
+Client types need to implement these methods to be used in a subquery:
+
+| Function Name | Purpose | Explanation |
+| ------------- | ------- | ----------- |
+| `getSubqueryValues` | To be used as a source node | Takes the new results for a node as input and should return a list of values (typically an array). Also takes the actual source node as the second parameter. |
+| `useSubqueryValues` | To be a target node | Takes a values list (the output of a getSubqueryValues call) and produces a changeset that is passed to a `mutate` action. Also takes the actual target node as the second parameter. |
```

Comment:
Little question: Could this be a `subquery` object with `input` and `output` instead? Or something like mongo's `from` and `to`.

Reply:
I'm not sure I follow. You specify how to get and use values on a per-type basis. It's not about an input/output object. We actual use from/to in the subquery signature...

Reply:
`get` and `use` is fine :)